### PR TITLE
PoC of rotating 3d object

### PR DESCRIPTION
**Pull request by mekiert-ocado**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,11 +11,13 @@
   },
   "dependencies": {
     "react": "^18.2.0",
-    "react-dom": "^18.2.0"
+    "react-dom": "^18.2.0",
+    "three": "^0.162.0"
```
Comment (mekiert-ocado) [inline package.json]: Do not use `^` in version.

Comment (mekiert-ocado) [reply package.json]: Please create file `.npmrc` (https://medium.com/@pmmanav/what-is-a-npmrc-file-e7bd40bff3f0) and set `save-exact` to true

File: public/Koparka.obj
Comment (mekiert-ocado) [inline public/Koparka.obj:1]: Please do not Polish

File: src/App.tsx
Comment (mekiert-ocado) [inline src/App.tsx:1]: Please move it to separate file

File: src/App.tsx [hunk 2]
```diff
@@ -1,35 +1,69 @@
...
-          Edit <code>src/App.tsx</code> and save to test HMR
-        </p>
-      </div>
-      <p className="read-the-docs">
-        Click on the Vite and React logos to learn more
-      </p>
-    </>
-  )
+
+class App extends Component {
```
Comment (mekiert-ocado) [inline src/App.tsx]: Use functional component instead

File: src/App.tsx [hunk 3]
```diff
@@ -1,35 +1,69 @@
...
+    let rotationSpeed = 0.01;
+    
+    renderer.setSize(window.innerWidth, window.innerHeight);
+    document.body.appendChild(renderer.domElement);
+      
+    const ambientLight = new THREE.AmbientLight( 0xffffff );
+    scene.add( ambientLight );
+
+
+    /////
```
Comment (mekiert-ocado) [inline src/App.tsx]:
```suggestion
    // TODO Temporary code. Remove it later. 
```

File: src/App.tsx [hunk 4]
```diff
@@ -1,35 +1,69 @@
...
+    const lineMaterial = new THREE.LineBasicMaterial({ color: 0x00ff00 });
+    const lineGeometry = new THREE.BufferGeometry();
+    lineGeometry.setFromPoints([new THREE.Vector3(0, -window.innerHeight / 2, 0), new THREE.Vector3(0, window.innerHeight / 2, 0)]);
+    const line = new THREE.Line(lineGeometry, lineMaterial);
+    scene.add(line);
+    /////
+
+    let loader = new OBJLoader();
```
Comment (mekiert-ocado) [inline src/App.tsx]: const

File: src/App.tsx [hunk 5]
```diff
@@ -1,35 +1,69 @@
...
+    loader.load(
+      '/Cerberus.obj',
+      function (object) {
+        object3D = object;
+        scene.add(object);
+      }
+    );
+      
+    function handleMouseMove(event: MouseEvent) {
+      const mouseX = (event.clientX / window.innerWidth) * 2 - 1;
```
Comment (mekiert-ocado) [inline src/App.tsx]: Please do not use magic numbers. Extract them to constants.

File: src/App.tsx [hunk 6]
```diff
@@ -1,35 +1,69 @@
...
+      rotationSpeed = mouseX * 0.03;
+    }
+
+    document.addEventListener('mousemove', handleMouseMove);
+
+    function animate() {
+    requestAnimationFrame(animate);
```
Comment (mekiert-ocado) [inline src/App.tsx]: indent

File: src/App.tsx [hunk 7]
```diff
@@ -1,35 +1,69 @@
-import { useState } from 'react'
+import React, { Component } from 'react';
+import * as THREE from 'three';
 import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
+import viteLogo from './vite.svg'
```
Comment (mekiert-ocado) [inline src/App.tsx]: It is not used.

File: src/vite.svg
Comment (mekiert-ocado) [inline src/vite.svg:1]: This file is not used. Remove it.

File: src/RotateObject3D.tsx [hunk 8]
```diff
@@ -0,0 +1,71 @@
+import * as THREE from 'three';
+import './App.css'
+import { OBJLoader } from 'three/addons/loaders/OBJLoader.js';
+import { MTLLoader } from 'three/addons/loaders/MTLLoader.js';
+
+  
+const RotateObject3D = () => {
```
Comment (mekiert-ocado) [inline src/RotateObject3D.tsx:8]: Use `useEffect` from React. 

https://legacy.reactjs.org/docs/hooks-effect.html

File: package.json (hunk 1 above)
Comment (BulldozerV8) [reply package.json]: I removed ^ and added file .nprmc with save-exact set to true

File: public/Koparka.obj
Comment (BulldozerV8) [reply public/Koparka.obj:1]: I changed every Polish words to English.

File: src/App.tsx
Comment (BulldozerV8) [reply src/App.tsx:1]: I created separeted file RotateObject3D and moved whole code to it

File: src/App.tsx (hunk 2 above)
Comment (BulldozerV8) [reply src/App.tsx]: Done

File: src/App.tsx (hunk 3 above)
Comment (BulldozerV8) [reply src/App.tsx]: I removed temporary code

File: src/App.tsx (hunk 4 above)
Comment (BulldozerV8) [reply src/App.tsx]: Done

File: src/App.tsx (hunk 5 above)
Comment (BulldozerV8) [reply src/App.tsx]: I declared variables for every number

File: src/App.tsx (hunk 6 above)
Comment (BulldozerV8) [reply src/App.tsx]: Done

File: src/App.tsx (hunk 7 above)
Comment (BulldozerV8) [reply src/App.tsx]: I removed unused files

File: src/vite.svg
Comment (BulldozerV8) [reply src/vite.svg:1]: Done

File: src/RotateObject3D.tsx (hunk 8 above)
Comment (BulldozerV8) [reply src/RotateObject3D.tsx:8]: I added useEffect and now the second object does not appear.

**Pull request merged**